### PR TITLE
python3Packages.gekko: init at 1.0.6

### PR DESCRIPTION
**Pull request by BatteredBunny**
## Description of changes

Added python [gekko](https://github.com/BYU-PRISM/GEKKO) package

## Things done

- Built on platform(s)
  - [x] x86_64-linux
  - [ ] aarch64-linux
  - [ ] x86_64-darwin
  - [ ] aarch64-darwin
- For non-Linux: Is sandboxing enabled in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
  - [ ] `sandbox = relaxed`
  - [ ] `sandbox = true`
- [ ] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://nixos.org/manual/nixpkgs/unstable/#sec-package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://nixos.org/manual/nixpkgs/unstable/#ssec-nixos-tests-linking) to the relevant packages
- [ ] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [x] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [23.11 Release Notes](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2311.section.md) (or backporting [23.05 Release notes](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2305.section.md))
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

### CODE REVIEW

**Review by katexochen — CHANGES_REQUESTED** [review]
Hey, currently your package isn't included anywhere, and thus won't be build. You must add it to `pkgs/top-level/python-packages.nix`.

File: pkgs/development/python-modules/gekko/default.nix [hunk 1]
```diff
@@ -0,0 +1,26 @@
+{
```
Comment (katexochen) [inline pkgs/development/python-modules/gekko/default.nix:1]: This should live in `pkgs/development/python-modules` instead.

**Comment by ayes-web** [comment]
> Hey, currently your package isn't included anywhere, and thus won't be build. You must add it to `pkgs/top-level/python-packages.nix`.

My bad, will fix it

File: pkgs/development/python-modules/gekko/default.nix (hunk 1 above)
Comment (ayes-web) [reply pkgs/development/python-modules/gekko/default.nix:1]: Not sure what you mean specifically but hopefully its fixed

Comment (katexochen) [reply pkgs/development/python-modules/gekko/default.nix:1]: nevermind, I was reviewing on mobile and misread the path of the file. :)

**Review by katexochen — COMMENTED** [review]
Sorry, I don't know much about Python in nixpkgs, but it looks like gekko would require numpy as dependency? https://github.com/BYU-PRISM/GEKKO/blob/master/setup.py#L89
Therefore I would have expected to find numpy in the list of `propagetedBuildInputs`. Is there some magic happening when fetching pypi?

Result of `nixpkgs-review pr 267470` run on x86_64-linux [1](https://github.com/Mic92/nixpkgs-review)
<details>
  <summary>4 packages built:</summary>
  <ul>
    <li>python310Packages.gekko</li>
    <li>python310Packages.gekko.dist</li>
    <li>python311Packages.gekko</li>
    <li>python311Packages.gekko.dist</li>
  </ul>
</details>

**Comment by ayes-web** [comment]
Your right it seems like it does need it, when i added `pythonImportsCheck` to the package it asked for it. Not sure why i got it working without but probably safest to add it

**Review by katexochen — CHANGES_REQUESTED** [review]
Code lgtm, but could you please format the code with nixpkgs-fmt?

```
$ nix shell nixpkgs#nixpkgs-fmt
$ nixpkgs-fmt pkgs/development/python-modules/gekko/default.nix
```

**Comment by ayes-web** [comment]
Yeah sure :)

**Review by katexochen — APPROVED** [review]

**Comment by emilytrau** [comment]
Thanks!

**Pull request merged**